### PR TITLE
Syncing upstream repo to merge changes 

### PR DESCRIPTION
**Pull request by dtamasang2**
## Changes proposed in this pull request:
- Syncing changes from `concourse/s3-resource` in order to remain unto date with changes to the repo
- README.md has been updated to reflect additional features

## Things to check

- For any logging statements, is there any chance that they could be logging sensitive data?
- Are log statements using a logging library with a logging level set? Setting a logging level means that log statements "below" that level will not be written to the output. For example, if the logging level is set to `INFO` and debugging statements are written with `log.debug` or similar, then they won't be written to the otput, which can prevent unintentional leaks of sensitive data.

## Security considerations
There are no security considerations for this change.

### CODE REVIEW

**Review by jameshochadel — APPROVED** [review]
Let's do the following: 

* Before merging, tag the latest commit on `main` with some tag so we can quickly find it in case we have to revert the sync
* After merging, delete the `CODEOWNERS` file, since the file from upstream includes a team member who isn't on our team 

I'm sure we'll have some follow-on work to reconcile their years of changes with ours.

**Comment by dtamasang2** [comment]
> Let's do the following:
> 
> * Before merging, tag the latest commit on `main` with some tag so we can quickly find it in case we have to revert the sync
> * After merging, delete the `CODEOWNERS` file, since the file from upstream includes a team member who isn't on our team
> 
> I'm sure we'll have some follow-on work to reconcile their years of changes with ours.

Thank you James for the feedback! That's a great idea. I have added the tag to the last commit id# b3270ef. I am still working on the merge conflicts. Thanks

**Pull request closed without merging**
